src/workerd/util/sqlite.c++

```diff
@@ -992,6 +998,8 @@ SqliteDatabase::Query::~Query() noexcept(false) {
       sqlite3_clear_bindings(&statement);
     }
   }
+  //Update the db stats that we have collected for the query
```

Comment:
Personally, I would do this before all of the reset stuff.  That way you know the `sqlite3_reset` doesn't affect your counts even without checking the implementation.

Reply:
I chose to do it here so we update the stats only once. The way the update is done here uses the internal variables of the query, so reset doesn't have an effect to any of the entities involved here

Reply:
I realize the confusion is arising from the updateQueryStats, the implementation is open to interpretation. I will rename it to addQueryStats, so it is clear that the stats will have to accumulated.

Reply:
I'm confused, I meant just move this block of code to the start of the dtor!

Reply:
Ah, I thought you wanted me to move this out of dtor and into the nextRow itself. 
Will do that! I still think the name was confusing, so I changed it